python/libcudf/pyproject.toml

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+[build-system]
+build-backend = "scikit_build_core.build"
+requires = [
+    "cmake>=3.26.4",
+    "cython>=3.0.3",
+    "ninja",
+    "numpy==1.23.*",
+    "pyarrow==14.0.2.*",
+    "rmm==24.6.*",
+    "scikit-build-core[pyproject]>=0.7.0",
```

Comment:
All we should need out of this list are cmake, ninja, and scikit-build-core. The updates should be made in dependencies.yaml, though, not here.

However, we will also need to add librmm and libkvikio as dependencies to this list. If you want to test that in CI, you'll have to do something like [this](https://github.com/rapidsai/raft/pull/2264/files#diff-f188362130c12de577292a252dbaf869013e7e79cdda87772b0a0f96b64758adR41).